### PR TITLE
`azurerm_data_factory_pipeline`: fix `headers` acceptable value

### LINKED ISSUE #24758: Error occurred unmarshalling JSON
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.0

### AzureRM Provider Version

3.90.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_pipeline

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory_pipeline" "powerbi_workspace_creation" {
  name            = "powerbi_workspace_creation"
  folder          = local.pbi_dp_folder
  data_factory_id = var.data_factory_id
  activities_json = templatefile(
    "${path.module}/../../templates/activities/common/pipelines/powerbi_workspace_pipeline_activities.json",
    {
      key_vault_linked_service = azurerm_data_factory_linked_service_key_vault.linked_service_key_vault.name
    }
  )
  variables = {
    "rapApiClientId"         = var.rap_api_clientid,
    "rapApiMatApplicationId" = var.rap_api_mat_application_id,
    "simsIdUrl"              = var.simsid_url,
    "rapApiUrl"              = var.rap_api_url,
    "reportingApiUrl"        = var.reporting_api_url,
    "azureKeyVaultUrl"       = var.azure_key_vault_url,
    "powerbiCapacityId"      = var.powerbi_capacityid,
    "resourceGroupName"      = var.databricks_resource_group_name,
    "wsSubscriptionId"       = var.azure_databricks_workspace_resource_id,
  }
}

Template json
[
  {
    "name": "get sims id token",
    "type": "WebActivity",
    "dependsOn": [
      {
        "activity": "get rap api client secret",
        "dependencyConditions": ["Succeeded"]
      }
    ],
    "policy": {
      "timeout": "0.00:05:00",
      "retry": 3,
      "retryIntervalInSeconds": 30,
      "secureOutput": false,
      "secureInput": true
    },
    "userProperties": [],
    "typeProperties": {
      "url": {
        "value": "@concat(variables('simsIdUrl'), 'connect/token')",
        "type": "Expression"
      },
      "method": "POST",
      "headers": {
        "Content-Type": "application/x-www-form-urlencoded"
      },
      "body": {
        "value": "@concat('client_id=',\nvariables('rapApiClientId'),'&client_secret=',\nactivity('get rap api client secret').output.value,'&grant_type=client_credentials','&scope=partner-read')",
        "type": "Expression"
      }
    }
  },
  {
    "name": "Get Sites by Application from RAP API",
    "type": "WebActivity",
    "dependsOn": [
      {
        "activity": "get sims id token",
        "dependencyConditions": ["Succeeded"]
      }
    ],
    "policy": {
      "timeout": "0.00:05:00",
      "retry": 3,
      "retryIntervalInSeconds": 30,
      "secureOutput": false,
      "secureInput": false
    },
    "userProperties": [],
    "typeProperties": {
      "url": {
        "value": "@concat(variables('rapApiUrl'),variables('rapApiMatApplicationId'),'/GetSiteByApplication')",
        "type": "Expression"
      },
      "method": "GET",
      "headers": {
        "authorization": {
          "value": "@concat('bearer',' ',activity('get sims id token').output.access_token)",
          "type": "Expression"
        },
        "content_type": "application/x-www-form-urlencoded"
      }
    }
  },
  {
    "name": "Filter Mats",
    "type": "Filter",
    "dependsOn": [
      {
        "activity": "Get Sites by Application from RAP API",
        "dependencyConditions": ["Succeeded"]
      }
    ],
    "userProperties": [],
    "typeProperties": {
      "items": {
        "value": "@json(activity('Get Sites by Application from RAP API').output.response)",
        "type": "Expression"
      },
      "condition": {
        "value": "@equals(coalesce(item().Authorised, 0), 1)",
        "type": "Expression"
      }
    }
  },
  {
    "name": "get rap api client secret",
    "type": "WebActivity",
    "dependsOn": [],
    "policy": {
      "timeout": "0.00:05:00",
      "retry": 3,
      "retryIntervalInSeconds": 30,
      "secureOutput": true,
      "secureInput": false
    },
    "userProperties": [],
    "typeProperties": {
      "url": {
        "value": "@concat(variables('azureKeyVaultUrl'), 'secrets/rap-api-client-secret/?api-version=7.0')",
        "type": "Expression"
      },
      "method": "GET",
      "authentication": {
        "type": "MSI",
        "resource": "https://vault.azure.net"
      }
    }
  },
  {
    "name": "create missing workspaces ForEach Mat",
    "type": "ForEach",
    "dependsOn": [
      {
        "activity": "Filter Mats",
        "dependencyConditions": ["Succeeded"]
      }
    ],
    "userProperties": [],
    "typeProperties": {
      "items": {
        "value": "@activity('Filter Mats').output.Value",
        "type": "Expression"
      },
      "isSequential": false,
      "batchCount": 2,
      "activities": [
        {
          "name": "create missing Workspace",
          "type": "WebActivity",
          "dependsOn": [],
          "policy": {
            "timeout": "0.00:05:00",
            "retry": 3,
            "retryIntervalInSeconds": 30,
            "secureOutput": false,
            "secureInput": false
          },
          "userProperties": [],
          "typeProperties": {
            "url": {
              "value": "@concat(variables('reportingApiUrl'), 'workspaces/', variables('powerbiCapacityId'))",
              "type": "Expression"
            },
            "method": "POST",
            "headers": {
              "Content-Type": "application/json"
            },
            "body": {
              "value": "@item()",
              "type": "Expression"
            },
            "authentication": {
              "type": "Basic",
              "username": "WorkspaceAdmin",
              "password": {
                "type": "AzureKeyVaultSecret",
                "store": {
                  "referenceName": "${key_vault_linked_service}",
                  "type": "LinkedServiceReference"
                },
                "secretName": "mat-frontend-webapi-basic-auth-secret"
              }
            }
          }
        }
      ]
    }
  }
]
```

### Debug Output/Panic Output

```shell
reading the state of Data Factory Pipeline "powerbi_workspace_creation": datafactory.PipelinesClient#Get: Failure responding to request: StatusCode=200 -- Original Error: Error occurred unmarshalling JSON - Error = 'json: cannot unmarshal object into Go struct field WebActivityTypeProperties.headers of type string' JSON = '{
```

### Expected Behaviour

Prior to the upgrade to azurerm 3.90.0 all datafactory pipelines where able to be read. Moving our azurerm provider back to 3.89.0 results in success again

### Actual Behaviour

reading the state of Data Factory Pipeline "powerbi_workspace_creation": datafactory.PipelinesClient#Get: Failure responding to request: StatusCode=200 -- Original Error: Error occurred unmarshalling JSON - Error = 'json: cannot unmarshal object into Go struct field WebActivityTypeProperties.headers of type string' JSON = '{

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #24758 

image

### CODE REVIEW

File: internal/services/datafactory/azuresdkhacks/pipelines.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/validation"
+	"github.com/tombuildsstuff/kermit/sdk/datafactory/2018-06-01/datafactory"
+)
+
+// TODO4.0: check if the workaround could be removed.
+// Workaround for https://github.com/hashicorp/terraform-provider-azurerm/issues/24758
+// Tracked on https://github.com/Azure/azure-rest-api-specs/issues/27816
+
```
Comment (magodo) [inline internal/services/datafactory/azuresdkhacks/pipelines.go:16]: If this file is just a copy paste of the original implementation, we shall also just write down something like: "This file is just a copy paste from file: ...".

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by magodo** [comment]
LGTM!

**Comment by tombuildsstuff** [comment]
@ziyeqf rather than doing this and needing to continue patching around this in the future - is `hashicorp/go-azure-sdk` usable as a replacement here? https://github.com/hashicorp/go-azure-sdk/blob/main/resource-manager/datafactory/2018-06-01/pipelines/method_createorupdate.go

**Comment by ziyeqf** [comment]
@tombuildsstuff I checked it, but it seems `go-azure-sdk` does not have the needed variant of `activity`
`go-azure-sdk`: https://github.com/hashicorp/go-azure-sdk/blob/4f74876e2154eae6add03690707943f92daad264/resource-manager/automation/2015-10-31/activity/model_activityproperties.go#L12
`kermit`: 
```
// WebActivity web activity.
type WebActivity struct {
	// WebActivityTypeProperties - Web activity properties.
	*WebActivityTypeProperties `json:"typeProperties,omitempty"`
	// LinkedServiceName - Linked service reference.
	LinkedServiceName *LinkedServiceReference `json:"linkedServiceName,omitempty"`
	// Policy - Activity policy.
	Policy *ActivityPolicy `json:"policy,omitempty"`
	// AdditionalProperties - Unmatched properties from the message are deserialized this collection
	AdditionalProperties map[string]interface{} `json:""`
	// Name - Activity name.
	Name *string `json:"name,omitempty"`
	// Description - Activity description.
	Description *string `json:"description,omitempty"`
	// State - Activity state. This is an optional property and if not provided, the state will be Active by default. Possible values include: 'ActivityStateActive', 'ActivityStateInactive'
	State ActivityState `json:"state,omitempty"`
	// OnInactiveMarkAs - Status result of the activity when the state is set to Inactive. This is an optional property and if not provided when the activity is inactive, the status will be Succeeded by default. Possible values include: 'ActivityOnInactiveMarkAsSucceeded', 'ActivityOnInactiveMarkAsFailed', 'ActivityOnInactiveMarkAsSkipped'
	OnInactiveMarkAs ActivityOnInactiveMarkAs `json:"onInactiveMarkAs,omitempty"`
	// DependsOn - Activity depends on condition.
	DependsOn *[]ActivityDependency `json:"dependsOn,omitempty"`
	// UserProperties - Activity user properties.
	UserProperties *[]UserProperty `json:"userProperties,omitempty"`
	// Type - Possible values include: 'TypeBasicActivityTypeActivity', 'TypeBasicActivityTypeExecuteWranglingDataflow', 'TypeBasicActivityTypeSparkJob', 'TypeBasicActivityTypeSynapseNotebook', 'TypeBasicActivityTypeScript', 'TypeBasicActivityTypeExecuteDataFlow', 'TypeBasicActivityTypeAzureFunctionActivity', 'TypeBasicActivityTypeDatabricksSparkPython', 'TypeBasicActivityTypeDatabricksSparkJar', 'TypeBasicActivityTypeDatabricksNotebook', 'TypeBasicActivityTypeDataLakeAnalyticsUSQL', 'TypeBasicActivityTypeAzureMLExecutePipeline', 'TypeBasicActivityTypeAzureMLUpdateResource', 'TypeBasicActivityTypeAzureMLBatchExecution', 'TypeBasicActivityTypeGetMetadata', 'TypeBasicActivityTypeWebActivity', 'TypeBasicActivityTypeLookup', 'TypeBasicActivityTypeAzureDataExplorerCommand', 'TypeBasicActivityTypeDelete', 'TypeBasicActivityTypeSQLServerStoredProcedure', 'TypeBasicActivityTypeCustom', 'TypeBasicActivityTypeExecuteSSISPackage', 'TypeBasicActivityTypeHDInsightSpark', 'TypeBasicActivityTypeHDInsightStreaming', 'TypeBasicActivityTypeHDInsightMapReduce', 'TypeBasicActivityTypeHDInsightPig', 'TypeBasicActivityTypeHDInsightHive', 'TypeBasicActivityTypeCopy', 'TypeBasicActivityTypeExecution', 'TypeBasicActivityTypeWebHook', 'TypeBasicActivityTypeAppendVariable', 'TypeBasicActivityTypeSetVariable', 'TypeBasicActivityTypeFilter', 'TypeBasicActivityTypeValidation', 'TypeBasicActivityTypeUntil', 'TypeBasicActivityTypeFail', 'TypeBasicActivityTypeWait', 'TypeBasicActivityTypeForEach', 'TypeBasicActivityTypeSwitch', 'TypeBasicActivityTypeIfCondition', 'TypeBasicActivityTypeExecutePipeline', 'TypeBasicActivityTypeContainer'
	Type TypeBasicActivity `json:"type,omitempty"`
}
```

File: internal/services/datafactory/azuresdkhacks/pipelines.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+					{
+						Target: "pipeline.Pipeline.Concurrency", Name: validation.Null, Rule: false,
+						Chain: []validation.Constraint{{Target: "pipeline.Pipeline.Concurrency", Name: validation.InclusiveMinimum, Rule: int64(1), Chain: nil}},
+					},
+				},
+			}},
+		},
+	}); err != nil {
+		return result, validation.NewError("datafactory.PipelinesClient", "CreateOrUpdate", err.Error())
+	}
```
Comment (tombuildsstuff) [inline internal/services/datafactory/azuresdkhacks/pipelines.go:63]: can we remove the validation here?

**Review by katbyte — APPROVED** [review]
Thanks @ziyeqf - LGTM 🎧

**Pull request merged**